messages/proto/messages.proto

```diff
@@ -46,7 +46,7 @@ message Message {
 // PrePrepareMessage is the message for the PREPREPARE phase
 message PrePrepareMessage {
   // proposal is the actual data being proposed for consensus
-  bytes proposal = 1;
+  ProposedBlock proposal = 1;
```

Comment:
Why not change the `Backend` interface to also accept the round number, or the new proto type?
This concern is outlined in `Issue G`, and it can be resolved by this.

This is the method I'm talking about:
```go
// InsertBlock inserts a proposal with the specified committed seals
InsertBlock(proposal []byte, committedSeals []*messages.CommittedSeal)
```

Reply:
Refactored

[310e6ad](https://github.com/madz-lab/go-ibft/pull/10/commits/310e6adb27cec1cc65ef1661565f2cb12280812e)